src/account/AccountGraphLoader.ts

```diff
@@ -187,9 +187,11 @@ export default class AccountGraphLoader {
   public static async loadBatch(graphClient: GraphClient, pageSize: number, pageNumber: number) {
     const response = await graphClient.queryOrThrow<AccountsQueryResponse>(accountsQuery, {pageSize, pageNumber});
     const accounts = new Map<string, AccountData>();
-    response.accounts.forEach(async (account) => {
```

Comment:
You can use an `await Promise.all([])` here instead.

Reply:
My worry was the following: If we run everything in parallel through `Promise.all`, accounts with the same assets may trigger identical `getSettlementMarket` calls in parallel and won't leverage the cache. In reality, Apollo might be wiser than I am and wait for current call to finish.

Let's stick with simplicity, `Promise.all` it shall be.